hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -694,6 +695,36 @@ void prepareComputation(Bytes input, final UnaryOperator<byte[]> aliasResolver)
                                             encoder,
                                             evmEncoder,
                                             precompilePricingUtils));
+                            case AbiConstants.ABI_ID_HRC_ASSOCIATE -> checkFeatureFlag(
+                                    dynamicProperties.isHRCAssociateEnabled(),
+                                    () -> checkHRCToken(
+                                            ledgers.tokens().exists(tokenId),
```

Comment:
Is this check here just to avoid duplicating some validation inside the `{Associate,Dissociate}Precompile`'s?

Reply:
Yes because if the token is null in the precompile code, we assume that we are called from HTS and try to decode the bytecode.  This is similar to what is done for other functions like approve.  I did think about creating a new subclass of AbstractAssociatePrecompile but it seemed to be overkill for I was trying to achieve here.

Reply:
Aha! Thanks for the explanation.